ethers-providers/src/gas_oracle/mod.rs

```diff
@@ -0,0 +1,75 @@
+mod eth_gas_station;
+pub use eth_gas_station::EthGasStation;
+
+mod etherchain;
+pub use etherchain::Etherchain;
+
+mod etherscan;
+pub use etherscan::Etherscan;
+
+use ethers_core::types::U256;
+
+use async_trait::async_trait;
+use reqwest::Error as ReqwestError;
+use thiserror::Error;
+
+/// Various gas price categories. Choose one of the available
+#[derive(Debug)]
+pub enum GasCategory {
```

Comment:
Usually such enums should derive a full set `#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]`. First few are for ease of use, but Hash may be required by end users (cache make a hash map of all the gas options from time to time and then query it)